javascript/javascript2/week2/hyf-bay/main.js

```diff
@@ -0,0 +1,62 @@
+console.log('Script loaded');
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector('section.products ul');
+console.log(productsUl);
+
+function renderProducts(products) {
+    productsUl.innerHTML=''
+    products.forEach(product => {
+        const li = document.createElement('li');
+
+        let shipsToHTML = '';
+        product.shipsTo.forEach(country => shipsToHTML += `<li>${country}</li>`);
+
+        li.innerHTML = `
+            <ul>
+                <li>${product.name}</li>
+                <li>${product.price}</li>
+                <li>${product.rating}</li>
+                <ul class="ships-to">${shipsToHTML}</ul>
+            </ul>
+        `;
+        productsUl.appendChild(li);
+    });
+}
+
+renderProducts(products);
+
+const userInput = document.getElementById("inputName");
+const userOption = document.getElementById("selectOption");
+const userCountry = document.getElementById("countryOption");
+const userPriceRange = document.getElementById("start");
+
+userInput.addEventListener("keyup", ()=>{
+    const searchedProduct = document.getElementById("inputName").value.toLowerCase()
+    return searchedProduct ? renderProducts(products.filter(a=>a.name.toLowerCase().includes(searchedProduct))) : 
+    renderProducts(products)   
+})
```

Comment:
This could be simplified by using the `change` listener:
```js
userInput.addEventListener("change", (event) => {
    const searchedProduct = event.target.value.toLowerCase()
    return searchedProduct ? renderProducts(products.filter(a=>a.name.toLowerCase().includes(searchedProduct))) : 
    renderProducts(products)   
})
```

Reply:
okay